### PR TITLE
dotnet-user-secrets: add support for json output and piping json input

### LINKED ISSUE #168: Make tool more friendly in usage from other tooling.
_From @phenning on August 24, 2016 20:9_

For example, in order to display a grid, you would need to parse out the info: output from the results of dotnet user-secrets list. Ideally, this should just be a --json or similar switch which would just output the secrets as json.

Similarly, it would also be good to be able to pass in json via the set command, bonus if nested json could be passed in here and the tool would take care of flattening. 

_Copied from original issue: aspnet/UserSecrets#95_

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves #168.

This adds support for `dotnet user-secrets list --json` which will spit out JSON.
It also adds support for piping input to the "set" command, e.g. `cat secrets.json | dotnet user-secrets set`

cc @phenning @muratg @glennc @moozzyk

### CODE REVIEW

**Comment by natemcmaster** [comment]
Can someone review this?

File: src/Microsoft.Extensions.SecretManager.Tools/CommandContext.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    internal class CommandContext
```
Comment (muratg) [inline src/Microsoft.Extensions.SecretManager.Tools/CommandContext.cs]: I'm OK keeping this internal, though the usual pattern I believe is to keep the classes public, and indicate that they're internal via the namespace.

File: src/Microsoft.Extensions.SecretManager.Tools/Internal/SetCommand.cs [hunk 2]
```diff
@@ -11,41 +14,90 @@ internal class SetCommand : ICommand
         private readonly string _keyName;
         private readonly string _keyValue;
 
-        public static void Configure(CommandLineApplication command, CommandLineOptions options)
+        public static void Configure(ExtendedHelpCommandLineApplication command, CommandLineOptions options)
         {
             command.Description = "Sets the user secret to the specified value";
+            command.AdditionalHelpInfo = @"
+Additional Info:
+  This command will also handled piped input. Piped input is expected to be a valid JSON format.
```
Comment (muratg) [inline src/Microsoft.Extensions.SecretManager.Tools/Internal/SetCommand.cs]: nit: tpyo: handled -> handle

File: src/Microsoft.Extensions.SecretManager.Tools/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+dotnet-user-secrets
```
Comment (muratg) [inline src/Microsoft.Extensions.SecretManager.Tools/README.md]: 👍

**Pull request merged**